src/Analyzers/CSharp/Analysis/RemoveRedundantToStringCallAnalysis.cs

```diff
@@ -64,10 +64,30 @@ public static void Analyze(SyntaxNodeAnalysisContext context, in SimpleMemberInv
                 if (containingType.SpecialType == SpecialType.System_String)
                     return true;
 
-                if (invocationExpression.WalkUpParentheses().IsParentKind(SyntaxKind.Interpolation)
-                    && IsNotHidden(methodSymbol, containingType))
+                ExpressionSyntax expression = invocationExpression.WalkUpParentheses();
+                switch (expression.Parent)
                 {
-                    return true;
+                    case InterpolationSyntax:
+                        {
+                            if (IsNotHidden(methodSymbol, containingType))
+                                return true;
+
+                            break;
+                        }
+                    case BinaryExpressionSyntax { RawKind: (int)SyntaxKind.AddExpression } addExpression:
+                        {
+                            if (addExpression.Right == expression
+                                && semanticModel.GetTypeInfo(addExpression.Left, cancellationToken).Type?.SpecialType == SpecialType.System_String)
+                                return true;
+
+                            if (addExpression.Left == expression
+                                && semanticModel.GetTypeInfo(addExpression.Right, cancellationToken).Type?.SpecialType == SpecialType.System_String
+                                && (addExpression.Right.WalkDownParentheses() is not InvocationExpressionSyntax invocationExpression2
+                                    || semanticModel.GetMethodSymbol(invocationExpression2, cancellationToken).Name != "ToString"))
```

Comment:
I'm not sure how the condition `semanticModel.GetMethodSymbol(invocationExpression2, cancellationToken).Name != "ToString")` should work. I tried that is not hit in the newly added tests. Could you add a test for it?

Reply:
This is included in the test that I have added
```
string s3 = o.ToString() + o[|.ToString()|];
```

We cannot remove ToString() on both sides of the addition or we change the semantic meaning. So for this case only suggest it for the one on the right.

Reply:
Ok, that makes sense 👍.